File: pinyougou_parent/pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/FreightTemplateController.java
package com.pinyougou.manager.controller;
import java.util.List;
import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbFreightTemplate;
import com.pinyougou.sellergoods.service.FreightTemplateService;
import com.pinyougou.util.ObjectUtils;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/freightTemplate")
public class FreightTemplateController {

    @Reference
    private FreightTemplateService freightTemplateService;

    /**
     * 查询所有 (包含模糊查询和分页)
     * @param freightTemplate
     * @param pageNum
     * @param pageSize
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(@RequestBody(required = false) TbFreightTemplate freightTemplate, Integer pageNum, Integer pageSize){
        return freightTemplateService.listFreightTemplateByExample(freightTemplate,pageNum, pageSize);
    }

    /**
     * 删除方法
     * @param idList
     * @return
     */
    @RequestMapping("/removeFreightTemplate")
    public Result removeFreightTemplate(Long[] idList) {
        if (!ObjectUtils.notEmpty(idList)) {
            return new Result(false, "未选中要删除的条目");
        }
        try {
            return freightTemplateService.removeFreightTemplate(idList);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "未知的错误信息,请重试");
        }
    }

    /**
     * 修改对象
     * @param freightTemplate
     * @return
     */
    @RequestMapping("/updateFreightTemplate")
    public Result updateFreightTemplate(@RequestBody(required = false) TbFreightTemplate freightTemplate) {
        if (!ObjectUtils.notEmpty(freightTemplate)) {
            return new Result(false, "对象不存在");
        }
        try {
            return freightTemplateService.updateFreightTemplate(freightTemplate);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "未知的错误信息,请重试");
        }
    }

    /**
     * 根据id查询
     * @return
     */
    @RequestMapping("/findOne")
    public TbFreightTemplate findOne(Long id){
        return freightTemplateService.getFreightTemplateById(id);
	}
	
    /**
     * 新增数据
     * @param freightTemplate
     * @return
     */
    @RequestMapping("/saveFreightTemplate")
    public Result saveFreightTemplate(@RequestBody(required = false) TbFreightTemplate freightTemplate){
        if (!ObjectUtils.notEmpty(freightTemplate)) {
            return new Result(false, "对象不存在");
        }
        try {
            return freightTemplateService.saveFreightTemplate(freightTemplate);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "未知的错误信息,请重试");
        }
    }
    /**
     * 返回全部列表(已过时的列表查询方法)
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbFreightTemplate> findAll(){
        return freightTemplateService.findAll();
    }
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/controller/freightTemplateController.js
app.controller('freightTemplateController', function ($scope, $controller, freightTemplateService) {//控制器
    //引入父类控制器
    $controller('baseController', {$scope: $scope});
    /**
     * 发送查询请求,定义findPage方法(已过时)
     */
    $scope.findPage = function (pageNum, pageSize) {
        freightTemplateService.findPage(pageNum, pageSize).success(function (data) {
					  
            $scope.rows = data.rows;
            $scope.paginationConf.totalItems = data.total;//更新总记录数
        })
    };
	
    /**
     * 搜索 (升级版findPage)
	 * 读取列表数据绑定到表单中
     */
    $scope.search = function (pageNum, pageSize) {
        freightTemplateService.search(pageNum, pageSize, $scope.searchEntity).success(function (data) {
					  
            $scope.rows = data.rows;
            $scope.paginationConf.totalItems = data.total;//更新总记录数
        })
    };
	
    /**
     * 定义新增方法
     */
    $scope.save = function () {
        var object = null;
        if ($scope.freightTemplate.id) {
            object = freightTemplateService.updateFreightTemplate($scope.freightTemplate);//修改
        } else {
            object = freightTemplateService.saveFreightTemplate($scope.freightTemplate);//增加
        }
        object.success(function (data) {
					  
            if (data.success) {
					//重新查询
                $scope.reloadList();//重新加载
            } else {
                alert(data.message);
            }
        });
    };

    /**
     * 定义根据id查找
     */
    $scope.findOne = function (id) {
							   
        freightTemplateService.findOne(id).success(function (data) {
            $scope.freightTemplate = data;
        })
    };
	
    /**
     * 删除方法
     * @param idList
     */
    $scope.removeFreightTemplateList = function (idList) {
        if (confirm("是否删除?")) {
            freightTemplateService.removeFreightTemplateList(idList).success(function (data) {
                if (data.success) {
							   
                    $scope.reloadList();
                } else {
                    alert(data.message);
                }
            });
        }
    };

});
